pkg/driver_cryptocell_310/psa_cryptocell_310/cipher_chacha20.c

```diff
@@ -114,46 +116,53 @@ psa_status_t psa_cipher_chacha20_update(psa_cipher_chacha20_ctx_t *ctx,
     DEBUG("Peripheral ChaCha20 Cipher update");
     psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
 
-    if (output_size < input_length)
+    if (output_size < (((ctx->buffer_length + input_length) >> 6) << 6))
     {
```

Comment:
```suggestion
    if (output_size < (((ctx->buffer_length + input_length) >> 6) << 6)) {
```

Reply:
Why those bit shifts? Probably worth a comment (remember, with `/* ... */`)

Reply:
That is for the truncating of the buffersize to a multiple of 64B